src/mds/MDSMap.cc

```diff
@@ -1169,3 +1176,99 @@ void MDSMap::set_min_compat_client(ceph_release_t version)
   std::sort(bits.begin(), bits.end());
   required_client_features = feature_bitset_t(bits);
 }
+
+const std::bitset<MAX_MDS>& MDSMap::get_bal_rank_mask_bitset() const {
+  return bal_rank_mask_bitset;
+}
+
+void MDSMap::set_bal_rank_mask(std::string val)
+{
+  bal_rank_mask = val;
+  dout(10) << "set bal_rank_mask to \"" << bal_rank_mask << "\""<< dendl;
+}
+
+const bool MDSMap::check_special_bal_rank_mask(std::string val, bal_rank_mask_type_t type) const
+{
+  if ((type == BAL_RANK_MASK_TYPE_ANY || type == BAL_RANK_MASK_TYPE_ALL) && (val == "-1" || val == "all")) {
+    return true;
+  }
+  if ((type == BAL_RANK_MASK_TYPE_ANY || type == BAL_RANK_MASK_TYPE_NONE) && (val == "0x0" || val == "0")) {
+    return true;
+  }
+  return false;
+}
+
+void MDSMap::update_num_mdss_in_rank_mask_bitset()
+{
+  int r = -EINVAL;
+
+  if (bal_rank_mask.length() and !check_special_bal_rank_mask(bal_rank_mask, BAL_RANK_MASK_TYPE_ANY)) {
```

Comment:
I think we would still like to use `&&` :)

Reply:
Oh I see. I can change 'and' to '&&'.

Reply:
@vshankar It has been modified. Thanks for your comment.

Reply:
Thank _you_ for the work!